### ISSUE TITLE
useRefinementList not working as expected on a geopoint field

### ISSUE THREAD

**Issue by iamarora** [typesense/typesense-instantsearch-adapter#188]
## Description

The refine method on a geopoint field is giving an error.

## Steps to reproduce

```
const { refine } = useRefinementList({ attribute: 'location_geo' });
// Tried the below
refine('(-44.50, 170.29, 20 mi)')
```

## Expected Behavior

It should be able to refine results with lat, long and radius.

## Actual Behavior

The error is
`Error: 400 - Value of filter field location_geo: must be in the ([-44.50, 170.29], radius: 0.75 km, exact_filter_radius: 5 km) or ([56.33, -65.97, 23.82, -127.82], exact_filter_radius: 7 km) format.`

## Metadata

**Typesense Version**: 0.26.0.rc25

**OS**: Ubuntu

Ref: https://typesense-community.slack.com/archives/C01P749MET0/p1698234116317759

**Comment by hdw1219** [comment]
Hi! any eta on this issue?

**Comment by iamarora** [comment]
> Hi! any eta on this issue?

Hey, I ended up using [this](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/geo-search/react/) to get my use case working. Doesn't look like refinement lists work with geopoint fields.

**Comment by hdw1219** [comment]
> > Hi! any eta on this issue?
> 
> Hey, I ended up using [this](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/geo-search/react/) to get my use case working. Doesn't look like refinement lists work with geopoint fields.

Thank you very much.

I use this temporary solution for now:

  const finalLocation = location ? location : undefined;

  const finalRadius = finalLocation ? radius * 1000 : undefined;

    <InstantSearch searchClient={searchClient} indexName={'candidats'}>
      <Configure
        aroundLatLng={finalLocation}
        aroundRadius={finalRadius}
      />
      

finalLocation:  45.5764258, -73.5466107
finalRadius:  75000

**Comment by iamarora** [comment]
> > > Hi! any eta on this issue?
> > 
> > 
> > Hey, I ended up using [this](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/geo-search/react/) to get my use case working. Doesn't look like refinement lists work with geopoint fields.
> 
> Thank you very much.
> 
> I use this temporary solution for now:
> 
> const finalLocation = location ? location : undefined;
> 
> const finalRadius = finalLocation ? radius * 1000 : undefined;
> 
> ```
> <InstantSearch searchClient={searchClient} indexName={'candidats'}>
>   <Configure
>     aroundLatLng={finalLocation}
>     aroundRadius={finalRadius}
>   />
> ```
> 
> finalLocation: 45.5764258, -73.5466107 finalRadius: 75000

Yeah, I had tried the configure widget as well. That's good to apply the initial filter, but when the values change dynamically from the FE it wasn't filtering then, but the initial filter was working well with the Configure widget.
